File: Python/Part-1/PrimeNumber.py
"""
Problem: Check whether a number is prime.
Example:
    Input  : 7
    Output : Prime Number
Logic:
    - Check divisibility from 2 to n//2
"""

num = int(input("Enter number: "))

if num <= 1:
    print(f"{num} is not a prime number")
else:
    for i in range(2, num // 2 + 1):
        if num % i == 0:
            print(f"{num} is not a prime number")
            break
    else:
        print(f"{num} is a prime number")


"""
Output:
------
Enter number: 31
31 is a prime number

"""

File: Java/PrimeNumber.java
/*
Problem: Write a Java program to check whether a number is a prime number or not.

What is a Prime Number?
    A prime number is a number greater than 1 that has
    exactly two factors: 1 and itself.

Example:
    Input  : 11
    Output : Prime Number

    Logic Used:
    - Check divisibility from 2 to n/2
    - If divisible → not prime
    - If not divisible → prime
------------------------------------------------------------
*/

import java.util.*;
class PrimeNumber{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter number:");

        int num = sc.nextInt();
        int count = 0;

        if(num <= 1)  // special case
        {
            System.out.println(num + " is not a prime number");
            return;
        }

        for(int i = 2; i <= num/2; i++)
        {
            if(num % i == 0)
            {
                count++;
                break;
            }
        }

        if(count == 0)
        {
            System.out.println(num + " is a prime number");
        }
        else
        {
            System.out.println(num + " is not a prime number");
        }
    }
}

/*
Output:
-------
Enter number:
7
7 is a prime number
*/